data/lib/compat/compat.lua

```diff
@@ -1065,6 +1065,19 @@ function doSetItemText(uid, text)
 	else
 		item:removeAttribute(ITEM_ATTRIBUTE_TEXT)
 	end
+
+	if writer ~= "" then
```

Comment:
`if writer ~= nil then`

Reply:
we removed almost every nil from lua scripts, so I'd suggest:
```suggestion
	if writer then
```

same for date

Reply:
```suggestion
	if writer and writer ~= "" then
```